bindings/go/credentials/resolve_direct.go

```diff
@@ -0,0 +1,65 @@
+package credentials
+
+import (
+	"context"
+	"fmt"
+	"maps"
+
+	"ocm.software/open-component-model/bindings/go/runtime"
+)
+
+func (g *Graph) resolveDirect(ctx context.Context, identity runtime.Identity) (map[string]string, error) {
+	// Check for cancellation to exit early
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+	}
+
+	vertex, err := g.matchAnyNode(identity)
+	if err != nil {
+		return nil, err
+	}
+
+	// Leaf node: return the credentials directly.
+	creds, ok := g.getCredentials(vertex.ID)
+	if ok {
+		return creds, nil
+	}
+
+	// Non–leaf node: recursively resolve each child and merge the results.
+	node := identity.String()
+
+	result := make(map[string]string)
+	if len(vertex.Edges) != 1 {
+		return nil, fmt.Errorf("failed to resolve credentials for node %q: multiple outgoing edges that would be candidates: %v", vertex.ID, vertex.Edges)
+	}
+	for id := range vertex.Edges {
```

Comment:
```suggestion
	result := make(map[string]string)
	if len(vertex.Edges) != 1 {
		return nil, fmt.Errorf("failed to resolve credentials for node %q: multiple outgoing edges that would be candidates: %v", vertex.ID, vertex.Edges)
	}
	id := vertex.Edges[0]
```
The `for` loop feels weird after we check for `edges ==1`